realm/src/main/java/io/realm/android/RealmCursor.java

```diff
@@ -0,0 +1,776 @@
+    }
+
+    /**
+     * Calling this method will close this cursor. The original RealmResults will not be affected. Trying to access any data in
+     * the cursor after calling this is illegal an the behavior is undefined.
+     */
+    @Override
+    public void close() {
+        closed = true;
+        table = null; // Instead of closing the table, we just release it. Original RealmResults would also be closed otherwise.
+        realm.removeChangeListener(changeListener);
+        dataSetObservable.notifyInvalidated();
+    }
+
+    /**
+     * Return true if the cursor is closed
+     * @return true if the cursor is closed.
+     */
+    @Override
+    public boolean isClosed() {
+        return closed;
+    }
+
+    /**
+     * Register an observer that is called when changes happen to the content backing this cursor.
+     *
+     * @param observer the object that gets notified when the content backing the cursor changes.
+     * @see #unregisterContentObserver(ContentObserver)
+     */
+    public void registerContentObserver(ContentObserver observer) {
+        contentObservable.registerObserver(observer);
+    }
+
+    /**
+     * Unregister an observer that has previously been registered with this
+     * cursor via {@link #registerContentObserver}.
+     *
+     * @param observer the object to unregister.
+     * @see #registerContentObserver(ContentObserver)
+     */
+    public void unregisterContentObserver(ContentObserver observer) {
+        // cursor will unregister all observers when it close
+        if (!closed) {
+            contentObservable.unregisterObserver(observer);
+        }
+    }
+
+    /**
+     * Registers an observer which is called when changes happen to the contents
+     * of the this cursors data set, for example, when the data set is changed via
+     * {@link #requery()}, {@link #deactivate()}, or {@link #close()}.
+     *
+     * @param observer the object that gets notified when the cursors data set changes.
+     * @see #unregisterDataSetObserver(DataSetObserver)
+     */
+    @Override
+    public void registerDataSetObserver(DataSetObserver observer) {
+        dataSetObservable.registerObserver(observer);
+    }
+
+    /**
+     * Unregisters an observer which has previously been registered with this
+     * cursor via {@link #registerContentObserver}.
+     *
+     * @param observer the object to unregister.
+     * @see #registerDataSetObserver(DataSetObserver)
+     */
+    @Override
+    public void unregisterDataSetObserver(DataSetObserver observer) {
+        dataSetObservable.unregisterObserver(observer);
+    }
+
+    /**
+     * Realm doesn't support {@link ContentResolver}. Calling this will throw an exception.
+     *
+     * @throws UnsupportedOperationException
+     */
+    @Override
+    public void setNotificationUri(ContentResolver cr, Uri uri) {
+        throw new UnsupportedOperationException("Notification URIs are not supported by RealmCursor");
```

Comment:
Partly. The primary reason is that AbstractCursor is really tightly coupled to the SQLite implementation. So instead of working around that it was easier to only use the interface (although some of the method implementations has been copied from AbstractCursor).